backend/alembic/versions/2025_11_09_0841-b3d2a0414e18_create_broadcast_tables.py

```diff
@@ -0,0 +1,272 @@
+        sa.Column("created_by", sa.Integer(), nullable=False),
+        sa.Column(
+            "status",
+            sa.String(length=50),
+            server_default="pending",
+            nullable=False,
+        ),
+        sa.Column("queued_at", sa.DateTime(), nullable=True),
+        sa.Column(
+            "created_at",
+            sa.DateTime(),
+            server_default=sa.text("now()"),
+            nullable=False,
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(),
+            server_default=sa.text("now()"),
+            nullable=False,
+        ),
+        sa.ForeignKeyConstraint(["created_by"], ["users.id"]),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_index(
+        "idx_broadcast_messages_created_by",
+        "broadcast_messages",
+        ["created_by"],
+    )
+
+    # 4. Create broadcast_message_groups table (many-to-many)
+    op.create_table(
+        "broadcast_message_groups",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("broadcast_message_id", sa.Integer(), nullable=False),
+        sa.Column("broadcast_group_id", sa.Integer(), nullable=False),
+        sa.Column(
+            "created_at",
+            sa.DateTime(),
+            server_default=sa.text("now()"),
+            nullable=False,
+        ),
+        sa.ForeignKeyConstraint(
+            ["broadcast_message_id"],
+            ["broadcast_messages.id"],
+            ondelete="CASCADE",
+        ),
+        sa.ForeignKeyConstraint(
+            ["broadcast_group_id"], ["broadcast_groups.id"]
+        ),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint(
+            "broadcast_message_id",
+            "broadcast_group_id",
+            name="unique_broadcast_message_group",
+        ),
+    )
+    op.create_index(
+        "idx_broadcast_message_groups_message",
+        "broadcast_message_groups",
+        ["broadcast_message_id"],
+    )
+    op.create_index(
+        "idx_broadcast_message_groups_group",
+        "broadcast_message_groups",
+        ["broadcast_group_id"],
+    )
+
+    # 5. Create broadcast_recipients table (delivery tracking)
+    op.create_table(
+        "broadcast_recipients",
+        sa.Column("id", sa.Integer(), nullable=False),
+        sa.Column("broadcast_message_id", sa.Integer(), nullable=False),
+        sa.Column("customer_id", sa.Integer(), nullable=False),
+        sa.Column(
+            "status",
+            postgresql.ENUM(
+                "PENDING",
+                "QUEUED",
+                "SENDING",
+                "SENT",
+                "DELIVERED",
+                "READ",
+                "FAILED",
+                "UNDELIVERED",
+                name="deliverystatus",
+                create_type=False,  # Use existing deliverystatus enum
+            ),
+            server_default="PENDING",
+            nullable=False,
+        ),
+        sa.Column(
```

Comment:
Why we have this template_message_sid?

Reply:
Hi mas @dedenbangkit,

This does not store the WhatsApp template ID. It stores the SID of the confirmation message sent before the actual broadcast; I renamed it to `confirm_message_sid`.
https://github.com/akvo/agriconnect/blob/feature/38-broadcast-message-api/backend/alembic/versions/2025_11_09_0841-b3d2a0414e18_create_broadcast_tables.py#L195-L197

This field is used in the linked function. If Twilio successfully sends the confirmation template, `confirm_message_sid` will be populated to indicate that.
https://github.com/akvo/agriconnect/blob/feature/38-broadcast-message-api/backend/tasks/broadcast_tasks.py#L144